lib/mongo/auth/stringprep.rb

```diff
@@ -0,0 +1,66 @@
+require 'mongo/auth/stringprep/tables'
+
+module Mongo
+  module Auth
+    module StringPrep
+      extend self
+
+      def prepare(data, mappings, prohibited, options = {})
+        out = apply_maps(data, mappings)
+        normalize(out) if options[:normalize]
+        check_prohibited(out, prohibited)
+        check_bidi(out) if options[:bidi]
+        out
+      end
+
+      private
+
+      def apply_maps(data, mappings)
+        out = ''
+
+        data.each_char do |c|
+          out << mapping(c, mappings)
+        end
+
+        out
+      end
+
+      def check_bidi(out)
+        if out.any{ |c| table_contains?(Tables::C8, c) }
```

Comment:
if out is a string, each and any? cannot be used, you'd have to use #each_char instead.

Reply:
Ah, didn't realize that. Fixed!